ql/src/java/org/apache/hadoop/hive/ql/optimizer/SortedDynPartitionOptimizer.java

```diff
@@ -635,34 +638,18 @@ public ReduceSinkOperator getReduceSinkOp(List<Integer> partitionPositions, List
         }
       }
 
-      // if partition and bucket columns are sorted in ascending order, by default
-      // nulls come first; otherwise nulls come last
-      Integer nullOrder = order == 1 ? 0 : 1;
+      char nullOrder = NullOrdering.defaultNullOrder(order, parseCtx.getConf()).getSign();
       if (sortNullOrder != null && !sortNullOrder.isEmpty()) {
-        if (sortNullOrder.get(0) == 0) {
-          nullOrder = 0;
-        } else {
-          nullOrder = 1;
-        }
-      }
-
-      for (Integer ignored : keyColsPosInVal) {
-        newSortNullOrder.add(nullOrder);
+        nullOrder = NullOrdering.fromCode(sortNullOrder.get(0)).getSign();
```

Comment:
I wondered why we pick up only the first element of `sortOrder` and `sortNullOrder`. This question doesn't directly relate to the problem of this ticket

Reply:
This logic was introduced by the very first implementation of `SortedDynPartitionOptimizer`.
https://issues.apache.org/jira/browse/HIVE-6455
I think the `sortOrder` is not empty only if the target table has ordering specified. Unfortunately the jira doesn't mention anything about such tables.